### PR TITLE
[config] Sort Config Db When Saving

### PR DESCRIPTION
**Pull request by tahmed-dev**
#### What I did

Restored cfggen sort method of json data and used it to
sort config db before saving.

closes: https://github.com/Azure/sonic-buildimage/issues/6328 
signed-off-by: Tamer Ahmed <tamer.ahmed@microsoft.com>

#### How I did it
Restored cfggen sorting method

#### How to verify it
Tested on target

### CODE REVIEW

File: config/main.py [hunk 1]
```diff
@@ -86,9 +88,20 @@
 asic_type = None
 
 #
-# Breakout Mode Helper functions
+# Helper functions
 #
 
+# Sort nested dict
+def sortDict(data):
```
Comment (jleveque) [inline config/main.py:95]: Python function names should be `snake_case`. I do realize that the function below (`readJsonFile`) isn't following this convention either, but it should for consistency.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: config/main.py [hunk 2]
```diff
@@ -86,11 +88,22 @@
 asic_type = None
 
 #
-# Breakout Mode Helper functions
+# Helper functions
 #
 
+# Sort nested dict
+def sort_dict(data):
+    """Sort of data dict"""
```
Comment (jleveque) [inline config/main.py:96]: Docstring is not very informative. Please add a bit more detail of how it sorts.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: config/main.py [hunk 3]
```diff
@@ -86,11 +88,24 @@
 asic_type = None
 
 #
-# Breakout Mode Helper functions
+# Helper functions
 #
 
+# Sort nested dict
+def sort_dict(data):
+    """ Sort of 1st leve and 2nd level dict of data naturally by its key
```
Comment (jleveque) [inline config/main.py:96]: Typo: s/leve/level

**Review by jleveque — CHANGES_REQUESTED** [review]

**Review by jleveque — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@tahmed-dev thanks. Can it be taken to 202012 and 201911 where the issue found ?

**Comment by tahmed-dev** [comment]
> @tahmed-dev thanks. Can it be taken to 202012 and 201911 where the issue found ?

Thanks @liat-grozovik, Yes, it could be taken to 201911 and 202012 branches.

**Comment by qiluo-msft** [comment]
202012 cherry-pick is not clean. Please submit another PR on 202012 branch.

**Comment by dprital** [comment]
@tahmed-dev - please let me know what will be the PR for 202012. Thanks.

**Comment by yxieca** [comment]
@tahmed-dev Please submit separate PR for 202012 branch. change cannot be cherry-picked cleanly.

**Comment by dprital** [comment]
@tahmed-dev - What is the PR for 202012 ? Thanks!

**Comment by tahmed-dev** [comment]
> @tahmed-dev - What is the PR for 202012 ? Thanks!

[PR:1651](https://github.com/Azure/sonic-utilities/pull/1651) port this one to 202012. @dprital I am curious what this PR is blocking? If it blocking test code, I strongly believe the test code has to adapt and we don't need this PR in production.

**Comment by prsunny** [comment]
@tahmed-dev , is this a bug fix? just curious, why is it required for 201911?

**Comment by tahmed-dev** [comment]
> @tahmed-dev , is this a bug fix? just curious, why is it required for 201911?

This is no bug here. the unsorted file was an artifact of removing `natsort` from config engine which in turn got removed from 201911 for fast reboot time enhancement.

There is no loss of functionality for master, 202012 or 201911. The issue that was created did not provide details why this is an issue. Removing 201911 label.

**Comment by prsunny** [comment]
> > @tahmed-dev , is this a bug fix? just curious, why is it required for 201911?
> 
> This is no bug here. the unsorted file was an artifact of removing `natsort` from config engine which in turn got removed from 201911 for fast reboot time enhancement.
> 
> There is no loss of functionality for master, 202012 or 201911. The issue that was created did not provide details why this is an issue. Removing 201911 label.

Thanks Tamer

**Pull request merged**